packages/create-cloudflare/src/templateMap.ts

```diff
@@ -1,53 +1,368 @@
-	},
-	webFramework: {
-		label: "Website or web app",
-		handler: runPagesGenerator,
-	},
-	common: {
-		label: "Example router & proxy Worker",
-		handler: runWorkersGenerator,
-	},
-	scheduled: {
-		label: "Scheduled Worker (Cron Trigger)",
-		handler: (args) => runWorkersGenerator({ ...args, deploy: false }),
-	},
-	queues: {
-		label: "Queue consumer & producer Worker",
-		handler: (args) => runWorkersGenerator({ ...args, deploy: false }),
-	},
-	chatgptPlugin: {
-		label: `ChatGPT plugin`,
-		handler: (args) =>
-			runWorkersGenerator({
-				...args,
-				ts: true,
-			}),
-	},
-	openapi: {
-		label: `OpenAPI 3.1`,
-		handler: (args) =>
-			runWorkersGenerator({
-				...args,
-				ts: true,
-			}),
-	},
-	"pre-existing": {
-		label: "Pre-existing Worker (from Dashboard)",
-		handler: runWorkersGenerator,
-		hidden: true,
-	},
+// A template can have a number of variants, usually js/ts
+type VariantInfo = {
+	path: string;
+};
+
+type StaticFileMap = Record<string, VariantInfo>;
+
+export type FrameworkMap = Awaited<ReturnType<typeof getFrameworkMap>>;
+export type FrameworkName = keyof FrameworkMap;
+
+export const getFrameworkMap = async () => ({
+	angular: (await import("../templates/angular/c3")).default,
+	astro: (await import("../templates/astro/c3")).default,
+	docusaurus: (await import("../templates/docusaurus/c3")).default,
+	gatsby: (await import("../templates/gatsby/c3")).default,
+	hono: (await import("../templates/hono/c3")).default,
+	next: (await import("../templates/next/c3")).default,
+	nuxt: (await import("../templates/nuxt/c3")).default,
+	qwik: (await import("../templates/qwik/c3")).default,
+	react: (await import("../templates/react/c3")).default,
+	remix: (await import("../templates/remix/c3")).default,
+	solid: (await import("../templates/solid/c3")).default,
+	svelte: (await import("../templates/svelte/c3")).default,
+	vue: (await import("../templates/vue/c3")).default,
+});
+
+export const getTemplateMap = async () => {
+	return {
+		"hello-world": await import("../templates/hello-world/c3.json"),
+		// Dummy record
+		webFramework: { displayName: "Website or web app" },
+		common: await import("../templates/common/c3.json"),
+		scheduled: await import("../templates/scheduled/c3.json"),
+		queues: await import("../templates/queues/c3.json"),
+		chatgptPlugin: await import("../templates/chatgptPlugin/c3.json"),
+		openapi: await import("../templates/openapi/c3.json"),
+		"remote-template": { displayName: "From a remote git repository" },
+		"pre-existing": (await import("../templates/pre-existing/c3")).default,
+	} as unknown as Record<string, TemplateConfig>;
 };
+
+export const selectTemplate = async (args: Partial<C3Args>) => {
+	// If not specified, attempt to infer the `type` argument from other flags
+	if (!args.type) {
+		if (args.framework) {
+			args.type = "webFramework";
+		} else if (args.existingScript) {
+			args.type = "pre-existing";
+		} else if (args.template) {
+			args.type = "remote-template";
```

Comment:
NIT: this piece of code made me realize the inconsistent casing we have, we have `webFramework` in camelCase and `pre-existing` and `remote-template` in kebab-case, could we keep this consinstent?

If this is out of scope here we can create a chore issue and deal with it later 🙂

Reply:
Yup it's unfortunately inconsistent but also a breaking change. Will file an issue and solve in a separate PR.

Reply:
:smiling_face_with_tear::+1: